src/mixins/widget.js

```diff
@@ -0,0 +1,64 @@
+const defaultShouldUpdate = (previous, next) => previous !== next;
+
+export const createWidgetMixin = ({ connect }) => ({
+  inject: {
+    instantSearchInstance: {
+      name: 'instantSearchInstance',
+      default() {
+        const tag = this.$options._componentTag;
+        throw new TypeError(
+          `It looks like you forgot to wrap your Algolia search component
+          "<${tag}>" inside of an "<ais-index>" component.`
+        );
+      },
+    },
+  },
+  data() {
+    return {
+      state: null,
+    };
+  },
+  computed: {
+    widgetParams() {
+      const { widgetParams } = this.$options;
+
+      return Object.keys(widgetParams).reduce(
+        (acc, name) => ({
+          ...acc,
+          [name]: widgetParams[name].getValue(this),
+        }),
+        {}
+      );
+    },
+  },
+  watch: {
+    widgetParams(next, previous) {
+      const { widgetParams } = this.$options;
+
+      const shouldUpdateWidget = Object.keys(widgetParams).some(name => {
```

Comment:
this is the Object.keys I meant (could have used entries too). My point was that if the thing was already and array, it would be slightly easier to iterate over it, but not a big issue at all